Sources/Request/Request/Request.swift

```diff
@@ -117,67 +102,23 @@ public struct AnyRequest<ResponseType> where ResponseType: Decodable {
-        // Url
-        guard var components = URLComponents(string: params.children!.filter({ $0.type == .url })[0].value as! String) else {
-            fatalError("Missing Url in Request body")
-        }
-        
-        // Query, QueryParam
-        let query = params.children!.filter({ $0.type == .query }).reduce([QueryParam]()) { (prev, param) in
-            if param.children != nil {
-                return prev + (param.children as! [QueryParam])
-            } else {
-                return prev + ([param] as! [QueryParam])
-            }
-        }
-        components.queryItems = query.map { param in
-            return URLQueryItem(name: param.key!, value: (param.value as! String))
-        }
-        
-        // BUILD REQUEST
-        var request = URLRequest(url: components.url!)
-        let method = params.children!.filter({ $0.type == .method })
-        if method.count > 0 {
-            request.httpMethod = (method[0].value as! MethodType).rawValue
-        }
-        
-        // Headers, Header
-        let headers = params.children!.filter({ $0.type == .header }).reduce([HeaderParam]()) { (prev, param) in
-            if param.children != nil {
-                return prev + (param.children as! [HeaderParam])
-            } else if param is HeaderParam {
-                return prev + ([param] as! [HeaderParam])
-            }
-            return prev
-        }
-        headers.forEach { header in
-            request.addValue(header.value as! String, forHTTPHeaderField: header.key!)
+        let params = self.params.sorted(by: { $0 is Url || ($1 is SessionParam) })
+
+        guard params.first is Url else {
+            fatalError("Request should contain at least one Url")
         }
-        
-        // Body
-        let body = params.children!.filter({ $0.type == .body })
-        if body.count > 0 {
-            request.httpBody = body[0].value as? Data
+
+        var request = URLRequest(url: URL(string: "https://")!)
+        for param in params where !(param is SessionParam) {
+            param.buildParam(&request)
         }
         
         // Configuration
         let configuration = URLSessionConfiguration.default
-        let timeouts = params.children!.filter { $0.type == .timeout }
-        if timeouts.count > 0 {
-            for timeout in timeouts {
-                guard let (source, interval) = timeout.value as? (Timeout.Source, TimeInterval) else {
-                    fatalError("Invalid Timeout \(timeout)")
-                }
-                if source.contains(.request) {
-                    configuration.timeoutIntervalForRequest = interval
-                }
-                if source.contains(.resource) {
-                    configuration.timeoutIntervalForResource = interval
-                }
-            }
+        for config in params where config is SessionParam {
+            (config as? SessionParam)?.buildConfiguration(configuration)
         }
-        
-        
+
```

Comment:
If you make the changes to `CombinedParams`, then this can be simplified to:
```suggestion
        var request = URLRequest(url: URL(string: "https://")!)
        let configuration = URLSessionConfiguration.default
        
        rootParam.buildParam(&request)
        (rootParam as? SessionParam)?.buildConfiguration(configuration)
```